.github/workflows/periodic.yml

```
@@ -8,6 +8,39 @@ on:
 permissions: {}
 
 jobs:
+  internal-integration:
+    name: Internal Integration
+    if: github.repository_owner == 'submariner-io'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+
+      - name: Update internal submariner-io/* dependencies to latest
+        run: |
+          for dep in $(awk '!/module/ && /github.com.submariner-io/ { print $1 }' go.mod)
+            do go get ${dep}@devel
+          done
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04
+        with:
+          title: Update submariner-io/* dependencies to latest
+          body: |
+            This checks the current status of this repository against the latest version of all the Submariner projects it depends on.
+            If something fails, the failure should be investigated and at least tracked as an issue blocking the next release.
+            Since some CI only runs periodically, if on-PR CI passes it's still good to merge this update for full integration coverage.
+          commit-message: |
+            Update submariner-io/* dependencies to latest
+
+            This checks the current status of this repository against the latest
+            version of all the Submariner projects it depends on.
```

Sorry, I’m obviously not communicating clearly enough.

The commit message generated by the job needs to make sense as a commit message in the history of the project it goes into. Its role in the context of the job is irrelevant; the commit itself doesn’t “check the current status …”, that’s what the job does.

```suggestion
            Update submariner-io/* dependencies to latest

            This upgrades all our dependencies on other Submariner projects to their
            latest development snapshots, ensuring the code in the projects remains
            coherent and that tests of development images verify the latest code.
```

---

I see what you're saying, but to me it seems the commit does "check the status of this repo against the latest...". It causes tests to be run against those versions, both on-PR and periodic. I can change to the suggestion of course, that's more clear/verbose.

---

A commit message describes the changes that the commit makes. The only thing that the commit changes is a bunch of dependencies; there are no changes _in the commit_ which check the status of anything. Our CI means that because a PR is created with the commit, the status will be checked; that’s why it makes sense to say that in the _PR_ description.